src/main/java/com/chimbori/crux/articles/PostprocessHelpers.java

```diff
@@ -109,15 +109,17 @@ private static void replaceLineBreaksWithSpaces(Element topNode) {
   private static void removeTopLevelTagsNotLikelyToBeParagraphs(Element element) {
     for (Element childElement : element.children()) {
       if (!RETAIN_TAGS_TOP_LEVEL.contains(childElement.tagName())) {
-        Log.printAndRemove(childElement, "removeTopLevelTagsNotLikelyToBeParagraphs");
+        if (!isImportant(childElement))
+          Log.printAndRemove(childElement, "removeTopLevelTagsNotLikelyToBeParagraphs");
```

Comment:
Another way to keep this code cleaner is to modify `printAndRemove` to not remove if the attribute is present. Then all the call sites can stay unchanged.

Reply:
Speaking as a consumer of the code (as opposed to the author of it), I found this code easy to understand because the decisions of whether or not to keep a node are nicely centralized in `PreprocessHelpers` and `PostprocessHelpers`. That allowed me to come in and make changes with confidence. I think moving the check to `Log.printAndRemove()` would make the code harder to understand. I would not expect a method for logging to be applying additional business logic.

This is just my two cents. I'm happy to do whatever you'd like.

Reply:
Good idea, let’s leave this as-is.